packages/react/src/ReactHooks.js

```diff
@@ -93,6 +93,12 @@ export function useEffect(
   updateDeps?: Array<mixed> | void | null,
   destroy?: ((resource: {...} | void | null) => void) | void,
 ): void {
+  if (__DEV__ && create == null) {
+    console.warn(
+      'React Hook useEffect requires an effect callback. Did you forget to pass a callback to the hook?',
+    );
+  }
+
```

Comment:
We inline the body of the `if (__DEV__)` condition so it's fine to split them up like this:

```suggestion
  if (__DEV__ ) {
    if (create == null) {
      console.warn(
        'React Hook useEffect requires an effect callback. Did you forget to pass a callback to the hook?',
      );
    }
  }

```